### ISSUE TITLE
Naming of static routes should be unique by default

### ISSUE THREAD

**Issue by sjsadowski** [sanic-org/sanic#2678]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

After helping resolve the issue here: https://community.sanicframework.org/t/using-app-static-twice-reports-a-deprecation-warning/1146

Which stemmed from #2674 I believe that app.static() should have a unique name. All app.static() routes currently require a name="" parameter which defaults to "static"

### Describe the solution you'd like

It might make sense to infer the default name from the path if a name is not provided, or to generate a purely random one as the name itself is largely inconsequential if not being set explicitly.

**Comment by Tronic** [comment]
The file may be used explicitly in `url_for`, so it would have to be constructed from (web side) path then. This is a breaking change for anyone currently using name "static" and some other path than "/", where the new name would be `static.the.path.components.here`.

**Comment by Tronic** [comment]
Somewhat related is that of having multiple decorators on a single function, or having identically named separate handler functions within the same app (possibly in different modules), as those would also cause duplicate names if not manually specified. I have not checked but perhaps Blueprints already add their own part to the name, avoiding duplicate names when Blueprints are combined.

**Comment by ahopkins** [comment]
Yes blueprints behave this way, and yes this will happen as its about to be enforced.
